design_docs/protostar-architecture.md

```diff
@@ -0,0 +1,63 @@
+# Protostar modular architecture
+
+This document proposes the idea how to modularize Protostar. Goals of the modularization are
+- Allow to introduce better ownerships of the codebase parts
+- Make workflow asynchronous between smaller sub-teams operating on contracts between modules
+- Allow incremental rewrite to Rust
+- Remove compilation responsibility from Protostar codebase
```

Comment:
Won't we still have the compiler in Protostar CLI?

Reply:
Short-term yes, long term everything will be handled by scarb